modules/navigation/navigation_path_query_parameters_2d.cpp

```diff
@@ -0,0 +1,160 @@
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "navigation_path_query_parameters_2d.h"
+
+void NavigationPathQueryParameters2D::set_pathfinding_algorithm(const NavigationPathQueryParameters2D::PathfindingAlgorithm p_pathfinding_algorithm) {
+	switch (p_pathfinding_algorithm) {
+		case PATHFINDING_ALGORITHM_ASTAR: {
+			parameters.pathfinding_algorithm = gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR;
+		} break;
+		default: {
+			WARN_PRINT_ONCE("No match for used PathfindingAlgorithm - fallback to default");
+			parameters.pathfinding_algorithm = gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR;
+		} break;
+	}
+}
+
+NavigationPathQueryParameters2D::PathfindingAlgorithm NavigationPathQueryParameters2D::get_pathfinding_algorithm() const {
+	switch (parameters.pathfinding_algorithm) {
+		case gd::PathfindingAlgorithm::PATHFINDING_ALGORITHM_ASTAR:
+			return PATHFINDING_ALGORITHM_ASTAR;
+		default:
+			WARN_PRINT_ONCE("No match for used PathfindingAlgorithm - fallback to default");
+			return PATHFINDING_ALGORITHM_ASTAR;
+	}
+}
+
+void NavigationPathQueryParameters2D::set_map(const RID &p_map) {
+	parameters.map = p_map;
+}
+
+const RID &NavigationPathQueryParameters2D::get_map() const {
+	return parameters.map;
+}
+
+void NavigationPathQueryParameters2D::set_start_position(const Vector2 &p_start_position) {
+	start_position_vec2 = p_start_position;
+	parameters.start_position = Vector3(p_start_position.x, 0.0, p_start_position.y);
+}
+
+const Vector2 &NavigationPathQueryParameters2D::get_start_position() const {
+	return start_position_vec2;
+}
+
+void NavigationPathQueryParameters2D::set_target_position(const Vector2 &p_target_position) {
+	target_position_vec2 = p_target_position;
+	parameters.target_position = Vector3(p_target_position.x, 0.0, p_target_position.y);
+}
```

Comment:
Do you need a `target_position_vec2`? Can you not just construct the `Vector2` as required from the `Vector3`?

Reply:
This should no longer be required.